ui/app/adapters/kv/metadata.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import ApplicationAdapter from '../application';
+import { kvMetadataPath } from 'vault/utils/kv-path';
+
+export default class KvMetadataAdapter extends ApplicationAdapter {
+  namespace = 'v1';
+
+  _url(fullPath) {
+    return `${this.buildURL()}/${fullPath}`;
+  }
+
+  createRecord(store, type, snapshot) {
+    const { backend, path } = snapshot.record;
+    const id = kvMetadataPath(backend, path);
+    const url = this._url(id);
+    const data = this.serialize(snapshot);
+    return this.ajax(url, 'POST', { data }).then(() => {
+      return {
+        id,
+        data,
```

Comment:
Typically we spread `...data` here 🤔 - I can't think of how this could be an issue since we immediately re-request the metadata model. Just calling out if anyone foresees any side affects

Reply:
The serializer usually expects a `data` block, so I think this is correct 🤔 Maybe usually we spread within like:
```
return {
  data: {
    ...data,
    id
  }
}
```